src/Sylius/Bundle/UiBundle/Renderer/TwigComponentBlockRenderer.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UiBundle\Renderer;
+
+use Sylius\Bundle\UiBundle\Registry\TemplateBlock;
+use Symfony\UX\TwigComponent\ComponentRendererInterface;
+
+/** @internal */
+final class TwigComponentBlockRenderer implements TemplateBlockRendererInterface
+{
+    public function __construct(
+        private TemplateBlockRendererInterface $decoratedRenderer,
+        private ComponentRendererInterface $componentRenderer,
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $context
+     */
+    public function render(TemplateBlock $templateBlock, array $context = []): string
+    {
+        if (null === $templateBlock->getComponent()) {
+            return $this->decoratedRenderer->render($templateBlock, $context);
+        }
+
+        return $this->componentRenderer->createAndRender($templateBlock->getComponent(), ['context' => $templateBlock->getContext() + $context]);
```

Comment:
```suggestion
        return $this->componentRenderer->createAndRender(
            $templateBlock->getComponent(),
            ['context' => array_merge($templateBlock->getContext(), $context)],
        );
```
IMO it makes more sense to allow immediate overwriting of previously configured values via twig, also the tests for duplicated keys are missing.

Reply:
<img width="322" alt="CleanShot 2023-09-23 at 09 29 34@2x" src="https://github.com/Sylius/Sylius/assets/80641364/b8a85dcf-e540-4e06-a5b9-91e54d2483d5">
Config from the Twig is taken into account. I skipped the test for duplicated keys, as it's just how `+` between arrays works.

Reply:
Nvm, in this case, config's context takes precedence since the extension point is via the config and not directly in twig.